Enterprise-CMCS/macpro-mdct-seds
# Remove moment.js from SEDS

### Description
This PR removes the `moment` JS library from SEDS. We already have `date-fns`, and we don't need both.

We were using Moment<sup>1</sup> in one function, a date formatter, to do a time zone conversion. Time zone conversion can be done easily<sup>2</sup> these days with the JS built-in, `Intl.DateFormat`. So that's what we're using now.

While I was rewriting the formatter, I changed the date separator from `-` to `/`. Previously it displayed `month-day-year`; now it displays `month/day/year`. I also fixed a few bugs<sup>3</sup>, and added some tests.

This date formatter is used in three places in the app:
1. Quarterly reports table body: <img width="1363" alt="image" src="https://github.com/user-attachments/assets/6a871298-aae3-41df-bc4f-eb1cfc72ea91">
2. Form page footer: <img width="1215" alt="image" src="https://github.com/user-attachments/assets/b2cf1eaf-b62f-490d-9ec8-027bf617cee6">
3. Certification tab verbiage: <img width="1219" alt="image" src="https://github.com/user-attachments/assets/6e43dafa-0bfd-469b-a222-38ebe3b5d88d">

<sup>1. As you'll see, we weren't using `moment` directly, but its companion library `moment-timezone`. At any rate, both libraries are gone now.</sup>
<sup>2. At least, converting from UTC to a given zone is easy. Converting from a given zone to UTC is still tricky. Luckily, we're not doing that in SEDS.</sup>
<sup>3. The old formatter displayed the midnight hour as "0 am"; now we have "12 am". It displayed the noon hour as "12 am"; now we have "12 pm". It displayed the time zone as "EST" year-round; now we have "EDT" in the summer months. As matters of style we are also capitalizing AM/PM now, and trimming leading zeroes from the month and day.</sup>

### Related ticket(s)
CMDCT-3907

---
### How to test
1. Run SEDS locally
   * I recommend logging in as `adminuser@test.com`
2. Navigate to the forms for a specific state and quarter
   * I recommend choosing `Alabama`, then `2021`, then `Q1`
   * Note the correctly-formatted dates in the Last Updated column
   * You have have to adjust your screen width to see the complete date
   * I considered restyling this table, to display the complete date on narrower screens, but I anticipate replacing the entire table component soon, so I didn't want to spend any time on it.
3. Navigate into any form
   * I recommend form `21E`
   * Note the correctly-formatted date in the footer
4. Navigate to the form's Certification tab
   * Note the correctly-formatted status date in the tab content
   * Note also that this content will not appear unless the form's status is Provisional Certified or Final Certified.

### Notes
n/a

---
### Pre-review checklist


---
### Pre-merge checklist


#### Review

#### Security
_If either of the following are true, notify the team's ISSO (Information System Security Officer)._
